composition/composition.md

```diff
@@ -151,30 +159,63 @@ Now, I'm going to leave Component's instance method
 This is what Component's method returned: 473
 ```
 
-## A Longer Example: Product and ShoppingCart
+## One-to-One and One-to-Many Composition Relationships
+
+We just saw examples of one-to-one and one-to-many composition relationships.
+
+A one-to-one relationship is one where a composite class has an attribute single instance of a component class as in this example:
 
 ```python
-class Product:
+class ExampleComponent:
 
-    def __init__(self, price, quantity):
-        self.price = price
-        self.quantity = quantity
+    def __init__(self, name):
+        self.name = name
 
-    def calculate_cost(self):
-        return self.price * self.quantity
+class ExampleComposite:
 
-class ShoppingCart:
+    def __init__(self, name, component):
+        self.name = name
+        self.component = component
+```
 
-    def __init__(self, products):
-        self.products = products
+A one-to-many relationship is one where a composite class has an attribute that is a list of instances of a component class as in this example:
 
-    def calculate_total_cost(self):
-        total_price = 0.0
+```python
+class ExampleComposite:
 
-        for product in self.products:
-            total_price += product.calculate_cost()
+    def __init__(self, name, components):
+        self.name = name
+        self.components = components
+```
+
+Note that `components` is a list of `component` objects.
+
+## A Longer Example: Product and ShoppingCart
+
+Now we will look at one more example of a one-to-many composition relationship.
 
-        return total_price
+```python
+1 class Product:
+2 
+3    def __init__(self, price, quantity):
+4        self.price = price
+5        self.quantity = quantity
+6
+7    def calculate_cost(self):
+8        return self.price * self.quantity
+9
+10 class ShoppingCart:
+11
+12    def __init__(self, products):
+13        self.products = products
+14
+15    def calculate_total_cost(self):
+16        total_price = 0.0
+17
+18        for product in self.products:
+19            total_price += product.calculate_cost()
```

Comment:
Another alternative to numbering the lines could be to add an inline comment on the line you want to question about:
```suggestion
              total_price += product.calculate_cost()  # (A)
```
Then below, you could refer to this as the line marked with the (A) comment. This would also allow you to remove all the line numbering and not worry about alignment.